LLama/ChatSession.cs

```diff
@@ -134,6 +158,45 @@ public void SaveSession(string path)
         File.WriteAllText(historyFilepath, History.ToJson());
     }
 
+    /// <summary>
+    /// Get the session state.
+    /// </summary>
+    /// <returns>SessionState object representing session state in-memory</returns>
+    public SessionState GetSessionState()
+    {
+        return new SessionState(Executor.Context.GetState(), ((StatefulExecutorBase)Executor).GetStateData())
+        {
+            InputTransformPipeline = InputTransformPipeline,
+            OutputTransform = OutputTransform,
+            HistoryTransform = HistoryTransform,
+            History = History.ToJson()
+        };
+    }
+
+    /// <summary>
+    /// Load a session from a session state.
+    /// </summary>
+    /// <param name="state"></param>
+    /// <returns></returns>
+    /// <exception cref="ArgumentException"></exception>
+    public void LoadSession(SessionState state)
+    {
+        if (Executor is StatefulExecutorBase statefulExecutor)
+        {
+            statefulExecutor.LoadState(
+                JsonSerializer.Deserialize(
+                    state.ExecutorState, statefulExecutor.GetStateData().GetType()
+                ) as ExecutorBaseState ?? throw new ArgumentException("Executor state is invalid", nameof(state))
+            );
+        }
+        else
+        {
+            throw new ArgumentException("Executor must be a StatefulExecutorBase to support loading of session state", nameof(state));
+        }
+        Executor.Context.LoadState(state.ContextState);
+        History = ChatHistory.FromJson(state.History) ?? new();
```

Comment:
If the transforms are finally included in the session state, we should also load them here.

Reply:
Oh, forgot to do that, I'll correct it.